include/pybind11/pytypes.h

```diff
@@ -1004,10 +1004,27 @@ class capsule : public object {
     PYBIND11_OBJECT_DEFAULT(capsule, object, PyCapsule_CheckExact)
     PYBIND11_DEPRECATED("Use reinterpret_borrow<capsule>() or reinterpret_steal<capsule>()")
     capsule(PyObject *ptr, bool is_borrowed) : object(is_borrowed ? object(ptr, borrowed) : object(ptr, stolen)) { }
-    explicit capsule(const void *value, void (*destruct)(PyObject *) = nullptr)
```

Comment:
Maybe just deprecate this constructor for backward compatibility? (Just without the `= nullptr` default and `explicit`).